### PR TITLE
Unit Testing Walkthrough

### PR DESCRIPTION
**Pull request by samfalberg**
In these commits, I will be writing the query function for a calendar to find time slots for meetings. I will also be writing extra tests for said query function.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 1]
```diff
@@ -270,5 +274,141 @@ public void notEnoughRoom() {
...
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void justEnoughRoomWithOptional() {
+    // Have one atendee and one optional attendee, but make it so that there is just enough
```
Comment (dieseldesai) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:337]: attendee

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,72 @@
...
+    //Only add events that are shorter than the entire day
+    if (request.getDuration() < TimeRange.WHOLE_DAY.duration()) {
+
+        //If there are no events, or we're scheduling for people not attending, the whole day is free
+        if (events.isEmpty() || !isAttending) {
+            findMeeting.add(TimeRange.WHOLE_DAY);
+        } else {   //If there's a valid event, make time range from start til first event
+            findMeeting.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, startTimeEventsList.get(0).getWhen().start(), false));
+
+            //If there's more than one event, and they don't overlap, get the time between them
```
Comment (dieseldesai) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:59]: good use of comments to explain some complicated logic

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,72 @@
...
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> findMeeting = new ArrayList<TimeRange>();
+    
+    ArrayList<Event> eventsList = new ArrayList<Event>();
+    boolean isAttending = false;
+    for (Event event : events) {
+        eventsList.add(event);
+
+        //Check if anybody we're scheduling is attending any of the events
+        for (String attendent : request.getAttendees()) {
```
[the remark this answers is not in the document]
Comment (samfalberg) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:34]: I don't think I understand what the last line is doing. Can you explain?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/Event.java [hunk 4]
```diff
@@ -28,6 +29,26 @@
   private final TimeRange when;
   private final Set<String> attendees = new HashSet<>();
 
+  /**
+   * A comparator for sorting events by their start time in ascending order.
+   */
+  public static final Comparator<Event> ORDER_BY_START = new Comparator<Event>() {
```
Comment (kashiish) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/Event.java:35]: Good idea creating Comparators for the Event class!

**Pull request merged**